### PR TITLE
add proper casts/format specifiers

### PR DESCRIPTION
**Pull request by orilevari**
**Description**: SDL fixes for C6340 warnings

### CODE REVIEW

File: winml/lib/Api.Image/TensorToVideoFrameConverter.cpp [hunk 1]
```diff
@@ -428,8 +428,8 @@ void TensorToVideoFrameConverter::ConvertGPUTensorToDX12Texture(
...
   WINML_THROW_HR_IF_FALSE_MSG(E_INVALIDARG, tensorDesc.channelType != kImageTensorChannelTypeGRAY8 || tensorDesc.sizes[1] == 1, "Target tensor description expects kImageTensorChannelTypeGRAY8, but has %lld channels specified instead of 1.", tensorDesc.sizes[1]);
-  WINML_THROW_HR_IF_FALSE_MSG(E_INVALIDARG, tensorDesc.sizes[2] == outputDesc.Height, "Target tensor height (%lld) does not match input height (%d).", tensorDesc.sizes[2], outputDesc.Height);
-  WINML_THROW_HR_IF_FALSE_MSG(E_INVALIDARG, tensorDesc.sizes[3] == (UINT)outputDesc.Width, "Target tensor width (%lld) does not match input width (%d).", tensorDesc.sizes[3], (UINT)outputDesc.Width);
+  WINML_THROW_HR_IF_FALSE_MSG(E_INVALIDARG, tensorDesc.sizes[2] == outputDesc.Height, "Target tensor height (%lld) does not match input height (%lu).", tensorDesc.sizes[2], outputDesc.Height);
+  WINML_THROW_HR_IF_FALSE_MSG(E_INVALIDARG, tensorDesc.sizes[3] == (UINT)outputDesc.Width, "Target tensor width (%lld) does not match input width (%lu).", tensorDesc.sizes[3], (UINT)outputDesc.Width);
```
Comment (martinb35) [inline winml/lib/Api.Image/TensorToVideoFrameConverter.cpp:432]: why does width need the UINT cast but Height doesn't?

File: winml/lib/Api.Image/TensorToVideoFrameConverter.cpp [hunk 2]
```diff
@@ -716,8 +716,8 @@ void TensorToVideoFrameConverter::ConvertCPUTensorToSoftwareBitmap(
           tensorDesc.channelType == kImageTensorChannelTypeRGB8,
       "Target tensor description expects kImageTensorChannelTypeGRAY8, kImageTensorChannelTypeBGR8, or kImageTensorChannelTypeRGB8 but has %d was specified.",
       tensorDesc.channelType);
-  WINML_THROW_HR_IF_FALSE_MSG(E_INVALIDARG, tensorDesc.sizes[2] == (UINT)height, "Target tensor height (%lld) does not match input height (%d).", tensorDesc.sizes[2], (UINT)height);
-  WINML_THROW_HR_IF_FALSE_MSG(E_INVALIDARG, tensorDesc.sizes[3] == (UINT)width, "Target tensor width (%lld) does not match input width (%d).", tensorDesc.sizes[3], (UINT)width);
+  WINML_THROW_HR_IF_FALSE_MSG(E_INVALIDARG, tensorDesc.sizes[2] == (UINT)height, "Target tensor height (%lld) does not match input height (%d).", tensorDesc.sizes[2], height);
+  WINML_THROW_HR_IF_FALSE_MSG(E_INVALIDARG, tensorDesc.sizes[3] == (UINT)width, "Target tensor width (%lld) does not match input width (%d).", tensorDesc.sizes[3], width);
```
Comment (martinb35) [inline winml/lib/Api.Image/TensorToVideoFrameConverter.cpp:721]: is this inconsistent with the previous change? On line 431&432 you are using "%lu" and here "%d" - is there a reason they are different?

File: winml/lib/Api.Image/VideoFrameToTensorConverter.cpp [hunk 3]
```diff
@@ -321,8 +321,8 @@ void VideoFrameToTensorConverter::ConvertDX12TextureToGPUTensor(
...
   WINML_THROW_HR_IF_FALSE_MSG(E_INVALIDARG, tensorDesc.channelType != kImageTensorChannelTypeGRAY8 || tensorDesc.sizes[1] == 1, "Target tensor description expects kImageTensorChannelTypeGRAY8, but has %lld channels specified instead of 1.", tensorDesc.sizes[1]);
-  WINML_THROW_HR_IF_FALSE_MSG(E_INVALIDARG, tensorDesc.sizes[2] == inputDesc.Height, "Target tensor height (%lld) does not match input height (%d).", tensorDesc.sizes[2], inputDesc.Height);
-  WINML_THROW_HR_IF_FALSE_MSG(E_INVALIDARG, tensorDesc.sizes[3] == (UINT)inputDesc.Width, "Target tensor width (%lld) does not match input width (%d).", tensorDesc.sizes[3], (UINT)inputDesc.Width);
+  WINML_THROW_HR_IF_FALSE_MSG(E_INVALIDARG, tensorDesc.sizes[2] == inputDesc.Height, "Target tensor height (%lld) does not match input height (%lu).", tensorDesc.sizes[2], inputDesc.Height);
+  WINML_THROW_HR_IF_FALSE_MSG(E_INVALIDARG, tensorDesc.sizes[3] == (UINT)inputDesc.Width, "Target tensor width (%lld) does not match input width (%lu).", tensorDesc.sizes[3], (UINT)inputDesc.Width);
```
Comment (martinb35) [inline winml/lib/Api.Image/VideoFrameToTensorConverter.cpp:326]: similar uint cast question as above

File: winml/lib/Api.Image/TensorToVideoFrameConverter.cpp (hunk 1 above)
Comment (orilevari) [reply winml/lib/Api.Image/TensorToVideoFrameConverter.cpp:432]: Im matching the casting that is done in the second parameter (tensorDesc.sizes[3] == (UINT)outputDesc.Width). the reason width needs the cast is that width is 64 bit and height is 32 bit

**Review by snnn — APPROVED** [review]

**Pull request merged**